### PR TITLE
[EOS-19407] BgDelete Code changes for new multipart design.

### PR DESCRIPTION
**Pull request by SeagateChaDeepak**
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: server/s3_post_complete_action.cc [hunk 1]
```diff
@@ -934,7 +934,7 @@ void S3PostCompleteAction::startcleanup() {
       // mark old OID for deletion in overwrite case, this optimizes
       // backgrounddelete decisions.
       ACTION_TASK_ADD(S3PostCompleteAction::mark_old_oid_for_deletion, this);
-      ACTION_TASK_ADD(S3PostCompleteAction::delete_old_object, this);
+      // ACTION_TASK_ADD(S3PostCompleteAction::delete_old_object, this);
```
Comment (knrajnambiar76) [inline server/s3_post_complete_action.cc:937]: why this is commented ?

Comment (SeagateChaDeepak) [reply server/s3_post_complete_action.cc]: Accidentally left it commented. Changed it.

**Comment by SeagateChaDeepak** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/CentOS-7.9.2009/job/S3Server-PreMerge-Test/16/console

**Comment by SeagateChaDeepak** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/CentOS-7.9.2009/job/S3-Server/50/console

**Review by nileshgovande — APPROVED** [review]

**Review by knrajnambiar76 — APPROVED** [review]

**Pull request merged**